### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'decorators')

### ISSUE THREAD

**Issue by capaj** [typescript-rtti/typescript-rtti#54]
The callstack:
```
RTTI: Failed to build source file /home/capaj/oss/graphql-repos/typegql/src/domains/arg/error.ts: Cannot read properties of undefined (reading 'decorators') [please report]
TypeError: Cannot read properties of undefined (reading 'decorators')
 at MetadataEncoder.params (/home/capaj/oss/graphql-repos/typegql/node_modules/typescript-rtti/src/transformer/metadata-encoder.ts:299:39)
    at MetadataEncoder.class (/home/capaj/oss/graphql-repos/typegql/node_modules/typescript-rtti/src/transformer/metadata-encoder.ts:93:35)
    at /home/capaj/oss/graphql-repos/typegql/node_modules/typescript-rtti/src/transformer/metadata-emitter.ts:89:83
    at MetadataEmitter.collectMetadata (/home/capaj/oss/graphql-repos/typegql/node_modules/typescript-rtti/src/transformer/metadata-emitter.ts:38:23)
    at /home/capaj/oss/graphql-repos/typegql/node_modules/typescript-rtti/src/transformer/metadata-emitter.ts:87:41
    at MetadataEmitter.scope (/home/capaj/oss/graphql-repos/typegql/node_modules/typescript-rtti/src/transformer/metadata-emitter.ts:53:20)
    at MetadataEmitter.class (/home/capaj/oss/graphql-repos/typegql/node_modules/typescript-rtti/src/transformer/metadata-emitter.ts:86:21)
    at /home/capaj/oss/graphql-repos/typegql/node_modules/typescript-rtti/src/transformer/common/visitor-base.ts:62:44
    at visitNodes (/home/capaj/oss/graphql-repos/typegql/node_modules/typescript/lib/typescript.js:87646:48)
    at visitLexicalEnvironment (/home/capaj/oss/graphql-repos/typegql/node_modules/typescript/lib/typescript.js:87686:22)
RTTI: During outboard metadata collection for class ArgError: Cannot read properties of undefined (reading 'decorators')
RTTI: Failed to build source file /home/capaj/oss/graphql-repos/typegql/src/domains/
```

I get this running in jest, typescript-rtti 0.5.6.
This is the source code:

```ts
import { reflect } from 'typescript-rtti'
import { BaseError } from '../../services/error'

import { ITargetAndField } from './compiler'

export class ArgError extends BaseError {
  constructor(msg: string, ctx: ITargetAndField, argIndex: number) {
    const { target, fieldName } = ctx
    const paramNames = reflect(target).getOwnMethod(fieldName).parameterNames
    const paramName = paramNames[argIndex]
    const fullMsg = `@Type ${target.name}.${fieldName}(${paramName} <-------): ${msg}`
    super(fullMsg)
    this.message = fullMsg
  }
}

```

**Comment by rezonant** [comment]
Minimal viable repro:

```
import { reflect } from 'typescript-rtti';\
export class Foo {
  constructor() {
    reflect(undefined)
  }
}
```

Cause: ApiCallTransformer transforms the constructor (and thus detaches `node.parent`) when `reflect()` is used, `design:*` metadata support requires checking if the class has at least one decorator to determine whether to emit `design:paramtypes` on a constructor. The check cannot be completed because we cannot check the class declaration, since the constructor no longer has a reference to it.

**Comment by rezonant** [comment]
This should be fixed in 0.6.0 which is now released.
